### PR TITLE
[DebugInfo] Change lowering of SIL instructions to use line 0 when appropriate

### PR DESCRIPTION
**Pull request by felipepiovezan**
This patch performs two NFC refactors to make the code control flow legible, and then fixes an issue where incorrect line numbers causes poor debugging experience.

### CODE REVIEW

File: lib/IRGen/IRGenDebugInfo.cpp [hunk 1]
```diff
@@ -341,6 +341,8 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
   IRGenDebugInfoFormat getDebugInfoFormat() { return Opts.DebugInfoFormat; }
 
 private:
+  FileAndLocation computeLLVMLoc(const SILDebugScope *DS, SILLocation Loc);
```
Comment (adrian-prantl) [inline lib/IRGen/IRGenDebugInfo.cpp:344]: short Doxygen comment?

File: lib/IRGen/IRGenDebugInfo.cpp [hunk 2]
```diff
@@ -342,6 +342,8 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
 
 private:
   FileAndLocation computeLLVMLoc(const SILDebugScope *DS, SILLocation Loc);
+  FileAndLocation computeLLVMLocCodeView(const SILDebugScope *DS,
```
Comment (adrian-prantl) [inline lib/IRGen/IRGenDebugInfo.cpp:345]: and here

File: lib/IRGen/IRGenDebugInfo.cpp [hunk 3]
```diff
@@ -2613,11 +2615,24 @@ bool IRGenDebugInfoImpl::lineEntryIsSane(FileAndLocation DL,
 }
 #endif
 
+/// In CodeView, zero is not an artificial line location. We try to avoid those
+/// line locations near user code to reduce the number of breaks in the
+/// linetables.
```
Comment (adrian-prantl) [inline lib/IRGen/IRGenDebugInfo.cpp:2620]: this should be in the declaration I think

**Comment by felipepiovezan** [comment]
https://github.com/swiftlang/llvm-project/pull/9614
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Comment by adrian-prantl** [comment]
The actual change looks good!

**Comment by felipepiovezan** [comment]
addressed review comments

**Comment by felipepiovezan** [comment]
https://github.com/swiftlang/llvm-project/pull/9614
@swift-ci test

**Comment by felipepiovezan** [comment]
https://github.com/swiftlang/llvm-project/pull/9614
@swift-ci test linux platform

**Comment by felipepiovezan** [comment]
https://github.com/swiftlang/llvm-project/pull/9614
@swift-ci test

**Comment by felipepiovezan** [comment]
Fixed Linux tests

**Comment by felipepiovezan** [comment]
https://github.com/swiftlang/llvm-project/pull/9614
@swift-ci test

**Pull request merged**